src/sas/qtgui/Plotting/QRangeSlider.py

```diff
@@ -172,9 +172,9 @@ def inputChanged(self):
         """ Track the input linked to the x value for this slider and update as needed """
         self._get_q()
         self.y_marker = self.base.data.y[(np.abs(self.base.data.x - self.x)).argmin()]
-        self.update()
+        self.update(draw=True)
 
-    def update(self, x=None, y=None):
+    def update(self, x=None, y=None, draw=False):
```

Comment:
Instead of extending the scope of `update` you could just add `self.base.draw()` after each `update` call with `draw=True`.
This will make `update` method more consistent in terms of functionality but will increase code duplication slightly.
I prefer to pass as few (unrelated) arguments as possible to a method to have it as clear/testable as possible.
But it's more of a stylistic issue than functionality - your call :)